libraries/Microsoft.BotKit.Adapters.Slack/SlackBotWorker.cs

```diff
@@ -0,0 +1,302 @@
+                convRef.User.Id = userId;
+                convRef.User.Name = null;
+                convRef.ChannelId = "slack";
+
+                return this.ChangeContextAsync(convRef);
+            }
+            else
+            {
+                throw new Exception("Error creating IM channel: " + channel.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Switch a bot's context into a different channel.
+        /// After calling this method, messages sent with `bot.say` and any dialogs started with `bot.beginDialog` will occur in this new context.
+        /// </summary>
+        /// <param name="channelId">A Slack channel id, like one found in `message.channel`.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        /// <param name="userId">A Slack user id, like one found in `message.user` or in a `.<@mention>`</param>
+        public async Task<object> StartConversationInChannel(string channelId, string userId)
+        {
+            var convRef = new ConversationReference();
+            var activity = this.Config.Activity;
+
+            convRef.Conversation.Id = channelId;
+            (convRef.Conversation as dynamic).Team = (activity.Conversation as dynamic).Team; // TO-DO: replace 'as dynamic'
+            convRef.User.Id = userId;
+            convRef.User.Name = null;
+            convRef.ChannelId = "slack";
+
+            return await this.ChangeContextAsync(convRef);
+        }
+
+        /// <summary>
+        /// Switch a bot's context into a specific sub-thread within a channel.
+        /// After calling this method, messages sent with `bot.say` and any dialogs started with `bot.beginDialog` will occur in this new context.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        /// <param name="channelId">A Slack channel id, like one found in `message.channel`.</param>
+        /// <param name="userId">A Slack user id, like one found in `message.user` or in a '.@mention'.</param>
+        /// <param name="threadTs">A thread_ts value found in the `message.thread_ts` or `message.ts` field.</param>
+        public async Task<object> StartConversationInThread(string channelId, string userId, string threadTs)
+        {
+            var convRef = new ConversationReference();
+            var activity = this.Config.Activity;
+
+            convRef.Conversation.Id = channelId;
+            (convRef.Conversation as dynamic).Team = (activity.Conversation as dynamic).Team; // TO-DO: replace 'as dynamic'
+            (convRef.Conversation as dynamic).ThreadTS = threadTs; // TO-DO: replace 'as dynamic'
+            convRef.User.Id = userId;
+            convRef.User.Name = null;
+            convRef.ChannelId = "slack";
+
+            return await this.ChangeContextAsync(convRef);
```

Comment:
These awaits needs the magic "don't block, this is a library" appended to them.

Reply:
Feedback applied. Thanks!